core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2917,23 +2923,40 @@ private void getExpressionConditionsAsList(Condition condition, List<ExpressionC
         return users;
     }
 
+    /**
+     * Method to verify whether the filter operation is a single attribute filter scenario or multi attribute filter scenario
+     *
+     * @param sizeOfExpressionConditions How many sub parts does Expression Condition has (value will be equals to 1
+     *                                   for a single attribute filter)
+     * @return True if the operation is a single attribute filter.
+     */
+    private boolean isSingleAttributeFilterOperation(int sizeOfExpressionConditions) {
+
+        // When the size of expression condition is equal to 1, the request is a filter with a single attribute.
+        if (sizeOfExpressionConditions == 1)
+            return true;
+        return false;
```

Comment:
We can directly return sizeOfExpressionConditions == 1

Reply:
Fixed:  62b2b1a48c4e8ba125d76efb35a4fc8ee3da1d60